Build fails during snapshot to image fails with OUT_OF_BOUNDS

I get the following error:
```
$ make flash ESP32_ENTRY=../../lask/jag-demo/icons.toit ESP32_WIFI=toitwifi ESP32_PASSWORD=toitware ESP32_PORT=/dev/ttyACM0
mkdir -p build/esp32/
make -C toolchains/esp32 -s /home/lau/toitlang/toit/build/esp32/include/sdkconfig.h
Toolchain path: /home/lau/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
Toolchain version: esp-2021r1
Compiler version: 8.4.0
Python requirements from /home/lau/toitware/esp-idf/requirements.txt are satisfied.
(cd build/host && cmake ../.. -G Ninja -DCMAKE_BUILD_TYPE=Release)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lau/toitlang/toit/build/host
(cd build/host && ninja build_toitvm)
[1/1] Generating ../bin/toitvm_boot.snapshot
build/host/bin/toitc -w build/snapshot ../../lask/jag-demo/icons.toit -Dwifi.ssid= -Dwifi.password=
(cd build/host && ninja build_toitvm)
[1/1] Generating ../bin/toitvm_boot.snapshot
build/host/bin/toitvm tools/snapshot_to_image.toit build/snapshot build/esp32/esp32.image.s
EXCEPTION error. 
OUT_OF_BOUNDS
  0: ByteArray_.[]=            <sdk>/core/collections.toit:1482:5
  1: ByteOrder.put_uint8       <sdk>/binary.toit:108:20
  2: LittleEndian.put_uint.<block> <sdk>/binary.toit:386:9
  3: SmallInteger_.repeat      <sdk>/core/numbers.toit:1004:3
  4: LittleEndian.put_uint     <sdk>/binary.toit:385:21
  5: ByteOrder.put_uint32      <sdk>/binary.toit:257:5
  6: Memory.put_word           tools/image.toit:66:32
  7: ToitHeapObject.write_header_to_ tools/image.toit:874:16
  8: ToitString.write_internal_to_ tools/image.toit:1016:5
  9: ToitString.write_to_      tools/image.toit:1006:12
 10: ToitHeapObject.write_to.<block> tools/image.toit:860:26
 11: Map.get.<block>           <sdk>/core/collections.toit:2561:30
 12: Map.get.<block>           <sdk>/core/collections.toit:2548:42
 13: HashedInsertionOrderedCollection_.find_body_ <sdk>/core/collections.toit:2061:3
 14: HashedInsertionOrderedCollection_.find_ <sdk>/core/collections.toit:2040:12
 15: Map.get                   <sdk>/core/collections.toit:2548:15
 16: Map.get                   <sdk>/core/collections.toit:2559:12
 17: Heap.store                tools/image.toit:187:33
 18: ToitHeapObject.write_to   tools/image.toit:859:23
 19: ToitTable.write_element   tools/image.toit:697:35
 20: ToitSequence.write_elements.<block> tools/image.toit:677:18
 21: SmallInteger_.repeat      <sdk>/core/numbers.toit:1004:3
 22: ToitSequence.write_elements tools/image.toit:677:10
 23: ToitSequence.fill_into    tools/image.toit:669:5
 24: ToitProgram.write_to      tools/image.toit:607:14
 25: build_image               tools/image.toit:1168:11
 26: main                      tools/snapshot_to_image.toit:75:12
 27: __entry__.<lambda>        <sdk>/core/entry.toit:48:20


make: *** [Makefile:91: build/esp32/esp32.image.s] Error 1
```
This is the code https://github.com/lask/jag-demo/commit/540298df0bb4d9e3a05258a0ef6c2df87b38f886 (link to the commit that fails to build).

--- a/tools/image.toit
+++ b/tools/image.toit
@@ -901,7 +901,7 @@ class ToitArray extends ToitHeapObject:
     return to_encoded_address address
 
 class ToitByteArray extends ToitHeapObject:
-  static INTERNAL_SIZE_CUTOFF := Image.PAGE_BYTE_SIZE_32 / 2
+  static INTERNAL_SIZE_CUTOFF := Image.PAGE_BYTE_SIZE_32 >> 2
   static RAW_BYTE_TAG ::= 0
 
   static LAYOUT_INTERNAL /ObjectType ::= ObjectType --packed {  // The '--packed' shouldn't be necessary, but doesn't hurt.
@@ -963,7 +963,7 @@ class ToitByteArray extends ToitHeapObject:
     return to_encoded_address address
 
 class ToitString extends ToitHeapObject:
-  static INTERNAL_SIZE_CUTOFF := Image.PAGE_BYTE_SIZE_32 / 2
+  static INTERNAL_SIZE_CUTOFF := Image.PAGE_BYTE_SIZE_32 >> 2
 
   static LAYOUT_INTERNAL /ObjectType ::= ObjectType --packed {  // The '--packed' shouldn't be necessary, but doesn't hurt.
     // Inherited from HeapObject.

